src/app/app-models/components/add-model/add-model.component.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import * as types from '../../../app-interactive-map/types';
+import {AppState} from '../../../store/app.reducers';
+import {select, Store} from '@ngrx/store';
+import {Observable} from 'rxjs';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {AddModel} from '../../store/models.actions';
+import {Project} from 'src/app/projects/types';
+import {MatDialog, MatSnackBar} from '@angular/material';
+import {AddedModelComponent} from './added-model.component';
+
+@Component({
+  selector: 'app-loader',
+  templateUrl: './add-model.component.html',
+  styleUrls: ['./add-model.component.scss'],
+
+})
+
+export class AddModelComponent implements OnInit, OnDestroy {
+
+  public allSpecies: Observable<types.Species[]>;
+  public allProjects: Observable<Project[]>;
+  public model: types.DeCaF.Model;
+  public addModelForm: FormGroup;
+  public error: Observable<Boolean>;
+  public fileType = '.json';
+  public reactions: string[] = [];
+  public modelError = false;
+  public addedModel = false;
+  public loading = false;
+
+  constructor(
+    private store: Store<AppState>,
+    public fb: FormBuilder,
+    private dialog: MatDialog,
+    public snackBar: MatSnackBar) {
+    this.addModelForm = this.fb.group({
+      name: ['', Validators.required],
+      organism_id: ['', Validators.required],
+      project_id: ['', Validators.required],
+      model_serialized: ['', Validators.required],
+      default_biomass_reaction: ['', Validators.required],
+    });
+  }
+
+  ngOnInit(): void {
+    this.allSpecies = this.store.pipe(select((store) => store.shared.allSpecies));
+    this.allProjects = this.store.pipe(select((store) => store.shared.projects));
+    this.error = this.store.pipe(select((store) => store.models.error));
+    this.store.pipe(select((store) => store.shared.modelHeaders)).subscribe(() => {
+        if (this.addedModel) {
+          this.dialog.closeAll();
+          this.snackBar.openFromComponent(AddedModelComponent, {
```

Comment:
Since it's just some text, maybe easier to open from a string instead of creating an extra component for this